source/agora/consensus/protocol/Nominator.d

```diff
@@ -475,22 +473,22 @@ extern(D):
                 envelope.statement.slotIndex, last_block.header.height.value);
             return;  // slot was already externalized or envelope is too new
         }
+
         const PublicKey public_key = PublicKey(envelope.statement.nodeID[]);
         const Scalar challenge = SCPStatementHash(&envelope.statement).hashFull();
-        const Point V = Point(public_key[]);
-        if (!V.isValid())
+        if (!public_key.isValid())
         {
             log.trace("Invalid point from public_key {}", public_key);
             return;
         }
-        if (!verify(V, envelope.signature.toSignature(), challenge))
+        if (!verify(public_key, envelope.signature.toSignature(), challenge))
         {
-            log.trace("INVALID Envelope signature {} \nfor public key {} \n" ~
-                "envelope {}\nchallenge {}",
-                envelope.signature, public_key, scpPrettify(&envelope),
-                challenge.toString(PrintMode.Clear));
+            // If it fails signature verification, it might not originate from said key
```

Comment:
This signature is for a single validator so I do not understand the comment.

Reply:
Anyone can produce an envelope claiming to be `X` with an invalid signature. Our message should not *attribute* the error to this node because it hasn't proved its identity.

Reply:
ok now I see what you meant. I was thinking it was related to multi sig.